src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -19,7 +19,10 @@ public partial class Process : IDisposable
         private static volatile bool s_initialized;
         private static readonly object s_initializedGate = new object();
         private static readonly ReaderWriterLockSlim s_processStartLock = new ReaderWriterLockSlim();
+
+#if !TARGET_MACCATALYST && !TARGET_IOS && !TARGET_TVOS
```

Comment:
nit: is there a simpler way of saying that given target is from apple but it's not macOS?

Reply:
`src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems:    <IsiOSLike Condition="'$(TargetsMacCatalyst)' == 'true' or '$(TargetsiOS)' == 'true' or '$(TargetstvOS)' == 'true'">true</IsiOSLike>` ?

Reply:
Also, you can avoid the ifdefs in the .cs files by moving the `ConfigureTerminalForChildProcesses` support into a separate .cs file, defining partial methods for it here, and including the .cs file only for targets that need it.

Reply:
Another alternative to solve this problem is to define a dummy method in the shim to make it link succesfully. This strategy is used in number of places today, e.g. https://github.com/dotnet/runtime/blob/main/src/libraries/Native/Unix/System.Native/pal_interfaceaddresses.c#L571